components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp

```diff
@@ -126,88 +126,70 @@
         }
 
         if (Boolean.parseBoolean(request.getParameter(SAMLSSOUIConstants.ENABLE_ATTRIBUTE_PROFILE))) {
-
             String claimsCountParameter = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyCounter");
             if (claimsCountParameter != null && !"".equals(claimsCountParameter)) {
-                try {
-                    int claimsCount = Integer.parseInt(claimsCountParameter);
-                    for (int i = 0; i < claimsCount; i++) {
-                        String claim = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyName" + i);
-                        if (claim != null && !"".equals(claim) && !"null".equals(claim)) {
-                            String[] currentClaims = serviceProviderDTO.getRequestedClaims();
-                            boolean isClaimAlreadyAdded = false;
-                            for (String currentClaim : currentClaims) {
-                                if (claim.equals(currentClaim)) {
-                                    isClaimAlreadyAdded = true;
-                                    break;
-                                }
-                            }
-
-                            if (!isClaimAlreadyAdded) {
-                                serviceProviderDTO.addRequestedClaims(claim);
+                int claimsCount = Integer.parseInt(claimsCountParameter);
+                for (int i = 0; i < claimsCount; i++) {
+                    String claim = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyName" + i);
+                    if (claim != null && !"".equals(claim) && !"null".equals(claim)) {
+                        String[] currentClaims = serviceProviderDTO.getRequestedClaims();
+                        boolean isClaimAlreadyAdded = false;
+                        for (String currentClaim : currentClaims) {
```

Comment:
I was not talking about converting an array to a map and then using the resultant map upon this particular functionality. One should not consider that sort of options anyway. Instead, my suggestion was to have it as a map right from the point where the claims are read from somewhere and kept it within the runtime object model.